### PR TITLE
Strict mode + Pharmacy.tsx

### PR DESCRIPTION
**Pull request by marisbest2**
I started with just refactoring Pharmacy.tsx a bit to clean up the large `initialize` method. 

But while doing so, I realized we didn't have `strict` on in the patient app. So this is now a very large PR and needs a bunch of testing too

### CODE REVIEW

**Comment by jknapp25** [comment]
top ranked pharm appears before rest of list

Screen Shot 2024-04-10 at 4 00 02 PM

**Comment by jknapp25** [comment]
missing address and distance

image

**Comment by jknapp25** [comment]
toggling filter ux doesn't show spinner while new pharmacies are loading

**Comment by jknapp25** [comment]
no pharmacies loaded after location change

image

**Comment by jknapp25** [comment]
redirect to /status throws error

image

**Comment by jknapp25** [comment]
filtered 24 hours then clicked show more till out of range, hangs in loading
image

**Comment by jknapp25** [comment]
If getting location doesn't resolve, for example if someone's browser blocked using their location, then they manually type an address and modal closes. Then re-open it's still spinning. Maybe we're not canceling loading location on modal close?

image

**Comment by marisbest2** [comment]
The latter one was existing behavior, but I fixed it.

File: apps/patient/src/api/external.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export const geocode = async (address: string) => {
     } else {
       throw new Error('No results found for the provided address.');
     }
-  } catch (error) {
+  } catch (error: any) {
```
Comment (jknapp25) [inline apps/patient/src/api/external.ts:19]: i thiiiiiink it's `google.maps.GeocoderError`, but not sure

Comment (marisbest2) [reply apps/patient/src/api/external.ts:19]: If the line on line `17` fires, it'll just be `Error`

**Comment by jknapp25** [comment]
Can't leave /review on demo

https://github.com/Photon-Health/client/assets/3934326/2ebeb12c-cab4-440f-b737-958bc2a72d20

File: apps/patient/src/api/external.ts (hunk 1 above)
Comment (jknapp25) [reply apps/patient/src/api/external.ts:19]: ah 👍🏼

File: apps/patient/src/utils/general.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ export const getFulfillmentType = (
   param?: string
 ): ExtendedFulfillmentType => {
   // We don't have COURIER fulfillment type yet, so manually check for those
-  if (pharmacyId in COMMON_COURIER_PHARMACY_IDS) {
```
Comment (captDaylight) [inline apps/patient/src/utils/general.ts:47]: hm so this was never working?

Comment (marisbest2) [reply apps/patient/src/utils/general.ts:47]: Yeah I think it probably wasn't. `in` does a key check

File: apps/patient/src/views/Pharmacy.tsx [hunk 3]
```diff
@@ -47,56 +48,97 @@ const PHARMACY_SEARCH_RADIUS_IN_MILES = 25;
...
+  const [pageOffset, setPageOffset] = useState(0);
 
-  const isMultiRx = flattenedFills.length > 1;
+  const isMultiRx = useMemo(() => flattenedFills.length > 1, [flattenedFills.length]);
 
-  const enableMailOrder = !isDemo && orgSettings.mailOrderNavigate;
-  const enableTopRankedCostco = !isDemo && orgSettings.topRankedCostco;
-  const enableTopRankedWalgreens = !isDemo && orgSettings.topRankedWalgreens;
-  const containsGLP = flattenedFills.some((fill) => isGLP(fill.treatment.name));
+  const enableMailOrder = useMemo(
```
Comment (captDaylight) [inline apps/patient/src/views/Pharmacy.tsx:109]: useMemo here might be overkill https://blog.logrocket.com/when-not-to-use-usememo-react-hook/

"Don’t use useMemo if your operation is inexpensive"

Comment (marisbest2) [reply apps/patient/src/views/Pharmacy.tsx:109]: Hmmm why did I use it... Probably there was a reason but yeah you're right

File: apps/patient/src/views/Pharmacy.tsx [hunk 4]
```diff
@@ -458,17 +560,14 @@ export const Pharmacy = () => {
     if (isDemo) {
       setTimeout(() => {
         setPreferredPharmacyId(pharmacyId);
-        toast({
-          title: 'Set preferred pharmacy',
-          ...TOAST_CONFIG.SUCCESS
-        });
+        toast({ ...TOAST_CONFIG.SUCCESS, title: 'Set preferred pharmacy' });
```
Comment (captDaylight) [inline apps/patient/src/views/Pharmacy.tsx:563]: "Preferred pharmacy set"

File: apps/patient/tsconfig.base.json [hunk 5]
```diff
@@ -14,7 +14,8 @@
     "skipLibCheck": true,
     "skipDefaultLibCheck": true,
     "baseUrl": ".",
-    "paths": {}
+    "paths": {},
+    "strict": true
```
Comment (captDaylight) [inline apps/patient/tsconfig.base.json:18]: here it is...

**Comment by marisbest2** [comment]
We really need to refactor to make the demo piece stand somewhat separately. For another day

**Comment by jknapp25** [comment]
Looks like we're missing our closes and opens strings 

image

image

**Comment by jknapp25** [comment]
Navigated to /status on demo shows error

image

**Comment by marisbest2** [comment]
I thought i just fixed those... maybe not, one sec

**Comment by jknapp25** [comment]
no pharm on demo status page

image

File: apps/patient/src/views/Pharmacy.tsx [hunk 6]
```diff
@@ -47,56 +48,87 @@ const PHARMACY_SEARCH_RADIUS_IN_MILES = 25;
...
-    setSelectedId('');
-    setShowFooter(false);
-    setShowingAllPharmacies(false);
-  };
+  // Pharmacy results
+  const [topRankedPharmacies, setTopRankedPharmacies] = useState<EnrichedPharmacy[]>([]);
+  const [pharmacyResults, setPharmacyResults] = useState<EnrichedPharmacy[]>([]);
+  const allPharmacies = useMemo(() => {
+    const topRankedIds = topRankedPharmacies.map((p) => p.id);
+    const combined = [
```
Comment (jknapp25) [inline apps/patient/src/views/Pharmacy.tsx:122]: love it, this fixes current duplication issue

**Comment by captDaylight** [comment]
I'm seeing Pharm on the Demo, also the standard flow works well + reroute flow

**Comment by jknapp25** [comment]
seeing pharm now, but pharm missing `closes at #:## pm` text on status page

image

**Comment by captDaylight** [comment]
Ah @jknapp25 good eye, I missed that

**Review by jknapp25 — APPROVED** [review]
🚀

**Pull request merged**